src/OSPSuite.Core/Domain/Services/SimModelManager.cs

```diff
@@ -69,24 +87,28 @@ public SimulationRunResults RunSimulation(IModelCoreSimulation simulation, Simul
          }
       }
 
+      private async Task loadAndRunSimulationAsync(IModelCoreSimulation simulation, CancellationToken cancellationToken)
+      {
+         await doIfNotCanceledAsync(() => loadSimulationAsync(simulation), cancellationToken);
+         await doIfNotCanceledAsync(() => FinalizeSimulationAsync(_simModelSimulation), cancellationToken);
+         await doIfNotCanceledAsync(() => simulateAsync(cancellationToken), cancellationToken);
+      }
+
       private void loadSimulation(IModelCoreSimulation simulation)
       {
          var xml = CreateSimulationExport(simulation, _simulationRunOptions.SimModelExportMode);
          _simModelSimulation = CreateSimulation(xml);
       }
 
-      private DataRepository getResults(IModelCoreSimulation simulation)
+      private async Task loadSimulationAsync(IModelCoreSimulation simulation)
       {
-         if (_canceled)
-            return new DataRepository();
-
-         return _dataFactory.CreateRepository(simulation, _simModelSimulation);
+         await Task.Run(() => loadSimulation(simulation)); // No need for a token here, it's a lightweight operation
```

Comment:
@rwmcintosh @benjaperez1983 
I don't understand. Isn't the SimModelMAnager always started in a Task already? Aren't we creating too much parallelism? Why are we creating a new task to load the simulation ?

should loadAndRunSimulationAsync do all of it at once instead of havign 3 calls each creating a task